catalyst/README.md

```diff
@@ -153,158 +158,413 @@ whenItem(store.buttonParent)
+
+```html
+<button>Continue</button>
+```
+
+```js
+$(elem).text("Pre-order");
+// expected output: will change the button text to "Pre-order"
+```
+
 #### attr()
-#### closest()
+
+This will return the attribute on the specified selector. If a parameter is provided, it will set the attribute content for the specified selector.
+
+```html
+<button aria-label="Continue">Continue</button>
+```
+
+```js
+$(elem).attr({"aria-label": "Pre-order"});
+// expected output: will change the aria-label text to "Pre-order"
+```
+
 #### each()
+
+This is loop iteration function that requires a callback input function.
+
+```html
+<ul id="sidebar" >
+  <li>First</li>
+  <li>Second</li>
+  <li>Third</li>
+</ul>
+```
+
+```js
+$("li").each( elem => {
+  elem.text("test");
+});
+// expected output: will change all the li elements to have the text "test"
+```
+
 #### watch()
+
+This function will wait for changes on the specified selector before executing the callback input function.
+
+```html
+<ul id="sidebar" >
+  <li>First</li>
+  <li>Second</li>
+  <li>Third</li>
+</ul>
+```
+
+```js
+$('#sidebar').watch( () =>{
+  $("li").each( elem => {
+    elem.text("test");
+  }
+});
+// expected output: will change all the li elements to have the text "test" when the ul is modified in anyway
+```
+
 #### first()
+
+This function will return the first child tag on the specified selector.
+
+```html
+<ul id="sidebar" >
+  <li>First</li>
+  <li>Second</li>
+  <li>Third</li>
+</ul>
+```
+
+```js
+$('li').first().text("first thing")
+// expected output: will change the first li ("First") element to have the text "first thing"
+```
+
 #### last()
+
+This function will return the last child tag on the specified selector.
+
+```html
+<ul id="sidebar" >
+  <li>First</li>
+  <li>Second</li>
+  <li>Third</li>
+</ul>
+```
+
+```js
+$('li').last().text("last thing")
+// expected output: will change the last li ("Third") element to have the text "last thing"
+```
+
 #### firstDom()
+
+This function will return the first child Element tag on the specified selector.
+
+```html
+<ul id="sidebar" >
+  <li>First</li>
+  <li>Second</li>
+  <li>Third</li>
+</ul>
+```
+
+```js
+$('li').firstDom().textContent = "first thing";
+// expected output: will change the first li ("First") element to have the text "first thing"
+```
+
 #### lastDom()
 
+This function will return the last child Element tag on the specified selector.
+
+```html
+<ul id="sidebar" >
+  <li>First</li>
+  <li>Second</li>
+  <li>Third</li>
+</ul>
+```
+
+```js
+$('li').lastDom().textContent = "last thing";
+// expected output: will change the last li ("Third") element to have the text "last thing"
+```
+
+
 ### rule.$$('<instrument key>')
-The $$ function for rule allows 
-    
+
+The $$ function for rule allows for the selection of a particular instrument keyed tag. You will then be able to 
+apply other catalyst functions to the specified instrument keyed tag.
 
 ## future
```

Comment:
Is this to mark a section that needs to be written?

Reply:
Yes, I left it open to Richard to fill it in. I'm not sure what the eventual goal is here so I left it as is.